backend/routes.js

```diff
@@ -17,57 +17,95 @@ import {
-        // if server is given a protocol argument, enforce it,
-        // otherwise decide which protocol to use
-        // depending if there are certificates
-        */
-        const apiVersion = '1.0.0';
-        if (args.protocol === 'HTTP' || isEmpty(this.certs)) {
-            this.server = restify.createServer({version: apiVersion});
-            this.protocol = 'http';
-            this.domain = 'localhost';
-        } else if (args.protocol === 'HTTPS' || this.certs) {
-            this.server = restify.createServer(merge({version: apiVersion}, this.certs));
-            this.protocol = 'https';
-            this.domain = getSetting('CONNECTOR_HTTPS_DOMAIN');
-        } else {
-            Logger.log('Failed to start the server.');
+         * `args` is of form {protocol: 'HTTP', createCerts: true}
+         * `args` is used to control whether we want to start the server initiall as http or
+         * https and whether we want to create certs if none found if started as http.
+         * It's main use is to control the flow during tests.
+         */
+        this.apiVersion = '1.0.0';
+
+        // Always start the HTTP server and keep it running.
+        this.httpServer.port = parseInt(getSetting('PORT'), 10);
+        this.httpServer.server = restify.createServer({version: this.apiVersion});
+        this.httpServer.protocol = 'http';
+        this.httpServer.domain = 'localhost';
+
+        if (args.startHttps && !getSetting('IS_RUNNING_INSIDE_ON_PREM')) {
+            // Create certs if necessary when we have an approved user.
+            if (args.createCerts && isEmpty(getCerts())) {
+                const createCertificates = setInterval(() => {
+                    // Can't create until user was authenticated.
+                    if (!isEmpty(getSetting('USERS'))) {
+                        clearInterval(createCertificates);
+                        fetchAndSaveCerts();
+                    }
+                }, 500);
+            }
+            const startHTTPS = setInterval(() => {
+                if (!isEmpty(getCerts())) {
+                    clearInterval(startHTTPS);
+                    this.httpsServer.start();
+                }
+            }, 500);
         }
 
         this.queryScheduler = new QueryScheduler();
 
-        this.start = this.start.bind(this);
-        this.close = this.close.bind(this);
+        this.httpServer.start = this.start.bind(this);
+        this.httpsServer.start = this.startHttpsServer.bind(this);
+
+        this.httpServer.close = this.close.bind(this);
+        this.httpsServer.close = this.closeHttpsServer.bind(this);
     }
 
-    start() {
+    startHttpsServer() {
+        // Reference the new certs into the instance.
+        this.httpsServer.certs = getCerts();
+        // TODO: Should HTTPS port be a setting too?
+        this.httpsServer.port = parseInt(getSetting('PORT_HTTPS'), 10);
+        // Start the interval to renew the certifications.
+        setRenewalJob();
```

Comment:
we should skip this step if it's running in on-prem.

are there any other things that we need to do to make sure that this doesn't break when running inside on-prem?